test/core/client_channel/resolvers/binder_resolver_test.cc

```diff
@@ -0,0 +1,164 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string.h>
+
+#include <gtest/gtest.h>
+
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+#include <grpc/support/string_util.h>
+
+#include "src/core/ext/filters/client_channel/resolver_registry.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "test/core/util/test_config.h"
+
+// Registers the factory with `grpc_core::ResolverRegistry`. Defined in
+// binder_resolver.cc
+void grpc_resolver_binder_init(void);
+
+namespace {
+
+class BinderResolverTest : public ::testing::Test {
+ public:
+  BinderResolverTest() {
+    factory_ = grpc_core::ResolverRegistry::LookupResolverFactory("binder");
+  }
+  ~BinderResolverTest() override {}
+  static void SetUpTestSuite() {
+    grpc_init();
+    if (grpc_core::ResolverRegistry::LookupResolverFactory("binder") ==
+        nullptr) {
+      // Binder resolver will only be registered on platforms that support
+      // binder transport. If it is not registered on current platform, we
+      // manually register it here for testing purpose.
+      grpc_resolver_binder_init();
+      ASSERT_TRUE(grpc_core::ResolverRegistry::LookupResolverFactory("binder"));
+    }
+  }
+  static void TearDownTestSuite() { grpc_shutdown(); }
+
+  void SetUp() override { ASSERT_TRUE(factory_); }
+
+  class ResultHandler : public grpc_core::Resolver::ResultHandler {
+   public:
+    void ReturnResult(grpc_core::Resolver::Result /*result*/) override {}
```

Comment:
If you're not actually checking the result here, then how do you know that it's returning the right thing?

Reply:
I copied from `sockaddr_resolver_test` and `dns_resolver_test` and they didn't check the result. Let me add the code to check the result if other concerns in this PR is resolved